src/solvers/lowering/byte_operators.cpp

```diff
@@ -108,6 +108,40 @@ static struct_exprt bv_to_struct_expr(
   return struct_exprt{std::move(operands), struct_type};
 }
 
+/// Convert a bitvector-typed expression \p bitvector_expr to a union-typed
+/// expression. See \ref bv_to_expr for an overview.
+static union_exprt bv_to_union_expr(
+  const exprt &bitvector_expr,
+  const union_typet &union_type,
+  const endianness_mapt &endianness_map,
+  const namespacet &ns)
+{
+  const union_typet::componentst &components = union_type.components();
+
+  mp_integer max_width = 0;
+  typet max_comp_type;
+  irep_idt max_comp_name;
+
+  for(const auto &comp : components)
+  {
+    auto element_width = pointer_offset_bits(comp.type(), ns);
+
+    if(!element_width.has_value() || *element_width <= max_width)
+      continue;
+
+    max_width = *element_width;
+    max_comp_type = comp.type();
+    max_comp_name = comp.get_name();
+  }
+
+  exprt operand = nil_exprt{};
+
+  if(max_width > 0)
+    operand = bv_to_expr(bitvector_expr, max_comp_type, endianness_map, ns);
+
+  return union_exprt{std::move(max_comp_name), std::move(operand), union_type};
```

Comment:
`max_comp_name` may be uninitialised here, in which case I suspect we shouldn't be trying to raise to a `union_exprt` at all?

Reply:
There's the case of an empty union - comments added (and there isn't any risk of uninitialised values anymore).